File: frontend/src/types/message.ts
export interface Message {
  id: string;
  conversation_id: string;
  sender_id: string;
  content: string;
  is_read: boolean;
  created_at: string;
}

export interface ConversationAdBrief {
  id: string;
  title: string;
  images: Array<{ thumbnail_url: string }>;
}

export interface ConversationUserBrief {
  id: string;
  name: string;
  avatar_url: string | null;
}

export interface Conversation {
  id: string;
  ad: ConversationAdBrief;
  other_user: ConversationUserBrief;
  last_message: Message | null;
  unread_count: number;
  updated_at: string;
}

export interface ConversationDetail {
  conversation: Conversation;
  messages: Message[];
  total: number;
  page: number;
}

export interface SendMessageRequest {
  ad_id: string;
  conversation_id?: string;
  content: string;
}

File: backend/app/schemas/message.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class MessageCreate(BaseModel):
    ad_id: uuid.UUID
    conversation_id: uuid.UUID | None = None
    content: str = Field(min_length=1, max_length=2000)


class MessageResponse(BaseModel):
    id: uuid.UUID
    conversation_id: uuid.UUID
    sender_id: uuid.UUID
    content: str
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ConversationAdBrief(BaseModel):
    id: uuid.UUID
    title: str
    images: list[dict] = []

    model_config = {"from_attributes": True}


class ConversationUserBrief(BaseModel):
    id: uuid.UUID
    name: str
    avatar_url: str | None = None

    model_config = {"from_attributes": True}


class ConversationResponse(BaseModel):
    id: uuid.UUID
    ad: ConversationAdBrief
    other_user: ConversationUserBrief
    last_message: MessageResponse | None = None
    unread_count: int
    updated_at: datetime

    model_config = {"from_attributes": True}


class UnreadCountResponse(BaseModel):
    count: int
